### ISSUE TITLE
Polygon 0x0A deciphered !

### ISSUE THREAD

**Issue by aybe** [phoboslab/wipeout#21]
Hi !

In fact they are sprites (https://github.com/phoboslab/wipeout/blob/master/wipeout.js#L507) except they their height should not be raised as sprites.

White stripes are 0x0A:

![2015-11-07 19_28_26-unity personal 64bit - scene3 unity - wxx-rebirth - pc mac linux standalone](https://cloud.githubusercontent.com/assets/1537591/11016551/c27b1aba-8585-11e5-8260-840fa1de7b19.png)

:+1:

**Comment by phoboslab** [comment]
Sorry, I don't understand. What do you mean with "their height should not be raised as sprites"?

Can you explain the data format? The `0x0B` sprite type for instance is defined by

``` javascript
Struct.uint16('index'),
Struct.uint16('width'),
Struct.uint16('height'),
Struct.uint16('texture'),
Struct.uint32('color')
```

**Comment by aybe** [comment]
Let me explain,

On the screenshot I have simply interpreted them as being regular sprites, using the formula in https://github.com/phoboslab/wipeout/blob/master/wipeout.js#L517 which add `height/2` to their position.

But their position should be unchanged, according to this in-game screenshot:

![2015-11-07 21_40_47-retroarch _ mednafen psx v0 9 36 5 __ fps_ 59 7 __ frames_ 12605952](https://cloud.githubusercontent.com/assets/1537591/11017205/3d20794c-8598-11e5-97ff-19f212d68aeb.png)

So basically they're exactly the same except for that vertical position offset you've added for regular sprites (such as trees for instance).

:D

**Comment by aybe** [comment]
Sorry but a bit of confusion here !

Basically on regular sprites, you add `height/2`, for those you need to subtract `height/2`:

![2015-11-07 21_54_43-greenshot](https://cloud.githubusercontent.com/assets/1537591/11017282/3453c1a0-859a-11e5-820a-c64a5f72eef0.png)

**Just added it to the simulation in this PR : https://github.com/phoboslab/wipeout/pull/22**

---

**By the way, have you seen the progress on my WXX-Rebirth project ?? :D**

BallisticNG (https://github.com/bigsnake09/BallisticNG) author gave me a pretty good hand in regards to the sim !!! : https://www.youtube.com/watch?v=tnTPEOm29zw

There's still a lot to be done but the project is definitely not dead !!!

Basically now I'm trying to split scene into objects (for animating them mainly), and this is a bit tougher than expected as I couldn't find any clue on game data. There is this Index1 in PRM object but unfortunately it's of no use since they're not unique. So it's the time for hunting manually, and in this hunting I'm trying to solve those yet 'unsolved' mysteries.

I'll let you know whenever I find any new relevant info on game data.

Cheers :+1: :+1: :+1:

**Comment by phoboslab** [comment]
Awesome! I noticed that you could say 0x0A is pinned at the top, while 0x0B is pinned at the bottom - which kinda makes sense, because trees are standing (0x0B) while the lights are hanging (0x0A). So I named the polygon types `SPRITE_BOTTOM_ANCHOR` and `SPRITE_TOP_ANCHOR` respectively.

Thanks!

Btw.: WXX Rebirth looks really, really cool!

**Comment by aybe** [comment]
That definitely makes sense !

Thanks Dominic, basically without your first insight in the game internals, all of this would have never happened, so thank you :+1:  :+1:  :+1:
